pkg/node/address_linux.go

```diff
@@ -42,6 +41,11 @@ func firstGlobalAddr(intf string, preferredIP net.IP, family int, preferPublic b
 
 retryInterface:
 	addr, err := netlink.AddrList(link, family)
+	// Ensure stable order so that the same IP is chosen every time.
+	// More info in the issue GH-7637.
+	sort.SliceStable(addr, func(i, j int) bool {
```

Comment:
What does this sort accomplish? `netlink.AddrList` is filtered to a `link` already, meaning all the `LinkIndex` fields on each address will be the same, producing the same list that `netlink.AddrList()` returned to begin with, no?

Reply:
Scratch that, I see `netlink.AddrList` may get called with nil.